mitosheet/mitosheet/mito_dash/v1/spreadsheet.py

```diff
@@ -0,0 +1,202 @@
+            return self.get_all_json()
+
+        @callback(Output(self.mito_id, 'all_json', allow_duplicate=True), Input(self.mito_id, 'data'), prevent_initial_call=True)
+        def handle_data_change_data(df_in_json):
+            df = pd.read_json(StringIO(df_in_json))
+            self._set_new_mito_backend((df, ))
+            return self.get_all_json()
+        
+    def _set_new_mito_backend(self, args):
+        """
+        Called when the component is created, or when the input data is changed.
+        """
+        self.mito_frontend_key = get_random_id()
+        self.mito_backend = MitoBackend(*args)
+        self.responses = []
+        def send(response):
+            self.responses.append(response)
+        self.mito_backend.mito_send = send
+        
+            
+    def process_single_message(self):
+
+        # If we are already processing messages -- then wait until it is
+        if self.processing_messages:
+            while self.processing_messages:
+                time.sleep(0.1)
+            
+        # Otherwise, set the processing flag to true
+        self.processing_messages = True
+
+        # Process all the messages in the queue
+        try:
+            if not self.unprocessed_messages.empty():
+                value = self.unprocessed_messages.get()
+                self.mito_backend.receive_message(value)
+        except:
+            # Make sure we always set the processing flag to false
+            pass
+            
+        # Set the processing flag to false
+        self.processing_messages = False
+        
+        
+    def get_all_json(self):
+        return json.dumps({
+            **self.mito_backend.get_shared_state_variables(),
+            'responses_json': json.dumps(self.responses),
+            'key': self.mito_frontend_key
+        })
+    
+    def get_result(self):
+        return SpreadsheetResult(
+            dfs=self.mito_backend.steps_manager.dfs,
+            code=self.mito_backend.steps_manager.code(),
+        )
+    
+
+def spreadsheet_callback(
+    *_args,
+    input_id=None
+):
+    """
+    Provides a server-side callback relating the values of one or more `Output` items 
+    to a single spreadsheet with the given id.
+
+    Example usage:
+    ```
+    from mitosheet.mito_dash.v1 import Spreadsheet, spreadsheet_callback
+    from dash import Dash, html, Output
+
+    import pandas as pd
+
+    df = pd.DataFrame({'A': [1, 2, 3]})
+
+    app = Dash(__name__)
+
+    app.layout = html.Div([
+        Spreadsheet(df, id='mito-dash-wrapper'),
+        html.Div(id='output'),    
+    ])
+
+    @spreadsheet_callback(
+        Output('output', 'children'),
+        input_id='mito-dash-wrapper',
+    )
+    def update_output(spreadsheet_result):
+        dfs = spreadsheet_result.dfs()
+        return f'Output: {str(dfs)}'
+
+    
+    if __name__ == '__main__':
+        app.run_server(debug=True)
+    ```
+
+    """
+    if input_id is None:
```

Comment:
Is `input_id` standard dash naming? Maybe a more helpful name would be:
- `trigger_spreadsheet_id`
- `synched_spreadsheet_id`
- `input_spreadsheet_id`

Reply:
Let's wait for some Plotly feedback on this one.